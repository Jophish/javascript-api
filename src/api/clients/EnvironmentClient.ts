import gateway from '../http/gateway';
import Environment, { IEnvironment } from '../dtos/Environment';

export default class EnvironmentClient {
  /**
   * Lists all environments
   */
  list(): Promise<Environment[]> | Promise<Error> {
    return gateway.get('/environments').then((res: any) => {
      return res.objects.map(e => new Environment(e));
    });
  };
}