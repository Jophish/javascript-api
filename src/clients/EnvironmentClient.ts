import Client from './Client';
import Environment, { IEnvironment } from '../dtos/Environment';

export default class EnvironmentClient extends Client {
  /**
   * Lists all environments
   */
  list(): Promise<Environment[]> {
    return this.gateway.get('/environments').then((res: any) => {
      return res.map(e => new Environment(e));
    });
  };
}
