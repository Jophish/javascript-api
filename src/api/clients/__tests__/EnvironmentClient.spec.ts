jest.mock('../../http/gateway');

import EnvironmentClient from '../EnvironmentClient';

test('should be a class (function)', () => {
  expect(typeof EnvironmentClient).toBe('function');
});

test('instances should expose an API', () => {
  const environment = new EnvironmentClient();

  expect(environment.list).toBeDefined();
});

test('it should be able return a promise resolving to list the environments', () => {
  const environment = new EnvironmentClient();
  const listPromise = environment.list();

  expect(listPromise.then).toBeDefined();
  expect(typeof listPromise.then).toBe('function');  

  return listPromise.then((res) => {
    expect(res).toMatchSnapshot();
  });
});