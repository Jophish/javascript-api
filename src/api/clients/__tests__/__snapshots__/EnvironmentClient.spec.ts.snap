// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should be able return a promise resolving to list the environments 1`] = `
Array [
  Environment {
    "id": "7ee47a70-4498-11e7-84b7-126798e48ef6",
    "name": "QA",
  },
  Environment {
    "id": "8105acc0-4498-11e7-a90e-0ee709202212",
    "name": "LOAD",
  },
  Environment {
    "id": "830ca480-4491-11e7-84b7-126798e48ef6",
    "name": "STG",
  },
  Environment {
    "id": "8344a470-4491-11e7-84b7-126798e48ef6",
    "name": "PROD",
  },
]
`;
